snap/snapcraft.yaml

```diff
@@ -0,0 +1,94 @@
+name: jitsi
+version: "1.0"
+summary: Jitsi
+description: |
+  This is a snap of Jitsi
+
+grade: stable
+confinement: strict
+base: core18
+
+# might need certificate configuration
+
+parts:
+  jicofo:
+    plugin: dump
+    source: https://download.jitsi.org/jitsi/debian/jicofo_1.0-508-1_amd64.deb
+  jitsi-meet-prosody:
+    plugin: dump
+    source: https://download.jitsi.org/jitsi/debian/jitsi-meet-prosody_1.0.3729-1_all.deb
+  jitsi-meet-web:
+    plugin: dump
+    source: https://download.jitsi.org/jitsi/debian/jitsi-meet-web_1.0.3729-1_all.deb
+  jitsi-videobridge:
+    plugin: dump
+    source: https://download.jitsi.org/jitsi/debian/jitsi-videobridge_1126-1_amd64.deb
+  jitsi-meet-web-config:
+    plugin: dump
+    source: https://download.jitsi.org/jitsi/debian/jitsi-meet-web-config_1.0.3729-1_all.deb
+    override-pull: |
+      snapcraftctl pull
+      echo "jitsi" > etc/hostname
+      echo "127.0.0.1 localhost jitsi" > etc/hosts
+      #usr/share/jitsi-meet/scripts/install-letsencrypt-cert.sh
+
+    stage-packages:
+      - prosody
+      - lua-bitop
+      - lua-event
+      - lua-expat
+      - lua-filesystem
+      - lua-sec
+      - lua-socket
+      - lua5.2
+      - openjdk-8-jre-headless
+      - ca-certificates-java
+      - java-common
+
+  jitsi-scripts:
+    source: ./scripts
+    plugin: dump
+
+layout:
+  /etc/jitsi:
+    bind: $SNAP/etc/jitsi
+
+environment:
+  JVB_HOSTNAME: jitsi
+  JVB_PORT: 5347
+  JVB_SECRET:
```

Comment:
can we have this autogenerated someho? I see that it needs to be specified in 2 places, it would be nice if we could have it in a single place where both read it from

Reply:
Yes. Snapcraft supports install hooks. Hooks are scripts that you can run at different stages in the snap lifecycle, including installation. So perhaps one of the things you would run would be to auto-generate secrets.

https://snapcraft.io/docs/supported-snap-hooks